core/embed/rust/src/ui/model_mercury/component/instructions.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    strutil::TString,
+    ui::{
+        component::{text::TextStyle, Component, Event, EventCtx, Never},
+        constant::WIDTH,
+        geometry::{Alignment, Offset, Rect},
+        model_mercury::theme,
+        shape::{Renderer, Text},
+    },
+};
+
+/// Component showing a task instruction (e.g. "Swipe up") and optionally task
+/// description (e.g. "Confirm transaction") to a user. The component
+/// is typically placed at the bottom of the screen. The height of the provided
+/// area must be 18px (only instruction) or 37px (both description and
+/// instruction). The content and style of both description and instruction is
+/// configurable separatedly.
+pub struct Instructions<'a> {
+    area: Rect,
+    text_instruction: TString<'a>,
+    text_description: Option<TString<'a>>,
+    style_instruction: &'static TextStyle,
+    style_description: &'static TextStyle,
+}
+
+impl<'a> Instructions<'a> {
+    /// height for component with only instruction [px]
+    pub const HEIGHT_SIMPLE: i16 = 18;
+    /// height for component with both description and instruction [px]
+    pub const HEIGHT_DEFAULT: i16 = 37;
+
+    pub fn new<T: Into<TString<'a>>>(instruction: T) -> Self {
```

Comment:
Nit: Putting the bound in the impl block might make this code a bit more readable. Not quite sure why we usually put them there and not in the struct definition.

```rust
impl<'a, T> Instructions<'a>
where
    T: Into<TString<'a>>,
{
```

Reply:
I did try but it seems that this syntax is not supported. I tried what you suggested and also this:
```rust
impl<'a, T: Into<String<'a>>> Instructions<'a> {...}
```
but I got the same compilation error:
![image](https://github.com/trezor/trezor-firmware/assets/14001709/960f95d1-6770-40ce-b03a-21187b2f3a36)

Reply:
ChatGPT also didn't suggest a way:

> However, this syntax is not directly supported in Rust, as Rust requires specifying trait bounds for each method where they are needed, not at the impl block level.
A common workaround for this repetition is to use a type alias with the bound included, but unfortunately, Rust's type system doesn't allow specifying generic parameter bounds on type aliases in a way that would solve this problem directly.

Reply:
my bad, didn't know this doesn't work when T is just a function argument